src/Resolvers/Microsoft.NET.Sdk.WorkloadMSBuildSdkResolver/WorkloadPartialResolver.cs

```diff
@@ -0,0 +1,258 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Build.Framework;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using Microsoft.NET.Sdk.WorkloadManifestReader;
+using System.Collections.Immutable;
+
+#if NET
+using Microsoft.DotNet.Cli;
+#else
+using Microsoft.DotNet.DotNetSdkResolver;
+#endif
+
+#if USE_SERILOG
+using Serilog;
+#endif
+
+#nullable disable
+
+namespace Microsoft.NET.Sdk.WorkloadMSBuildSdkResolver
```

Comment:
300 lines without unit test coverage. We should be able to test cache save and take.